common/src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -374,6 +387,30 @@ class Neo4jQueryReadStrategy(filters: Array[Filter] = Array.empty[Filter],
       Cypher.node(primaryLabel, otherLabels.asJava).named(name)
     }
   }
+
+  override def createStatementForGDS(options: Neo4jOptions): String = {
+    val retCols = requiredColumns.map(column => getCorrectProperty(column, null))
+    // we need it in order to parse the field YIELD by the GDS procedure...
+    val (yieldFields, args) = Neo4jUtil.callSchemaService(options, jobId, filters,
+      { ss => (ss.struct().fieldNames, ss.inputForGDSProc(options.query.value)) })
```

Comment:
```suggestion
      { ss => (ss.structForGDS().fieldNames, ss.inputForGDSProc(options.query.value)) })
```
?

Reply:
No `inputForGDSProc` has a different purpose, we need it in order to compute the input parameters required by the procedure so we can map the `.option("gds.*", <value>)` to each one and eventually complain about some missing field if the user does not provide it.

Reply:
For nodes and relationships we does not it and `inputFor*`